### PR TITLE
Add husky comit hook

### PR DESCRIPTION
**Pull request by LinnJS**
## Description

When opening PR https://github.com/department-of-veterans-affairs/vets-json-schema/pull/494 I found there was unintended changes that had not reached the dist possibly from previous commits. So I opened a PR with just `yarn build` again but added `yarn lint:fix && yarn build` in a husky pre-commit hook to hopefully prevent this kind of thing.

Add husky with pre-commit hooks for build and lint

image

### CODE REVIEW

**Comment by rianfowler** [comment]
Can we put the husky changes in a separate PR. I like the idea of adding some linting to this repo but I think we should consider that separately from the schema changes.

**Comment by rianfowler** [comment]
What do we want to do with this PR?

**Comment by ericboehs** [comment]
@LinnJS Any update here? Can we close/move back to draft?

**Comment by LinnJS** [comment]
If we can get some support for this update, I will go through and fix all of the errors currently caught by our rules.

I don't have any idea why people would reject this update. I think its important to enforce rules uniformly for code cleanliness like we do on all our other repos.

image

**Review by kelsonic — APPROVED** [review]
Standardization ftw 🚀 We should also add the linter as a check in CI

File: package.json [hunk 1]
```diff
@@ -13,7 +13,16 @@
...
     "lint:fix": "eslint src -c .eslintrc --fix --ext js",
-    "lint:changed": "LIST=`git diff-index --name-only HEAD | grep \"\\.js[x]\\{0,1\\}$\"`; if [ \"$LIST\" ]; then eslint --fix --quiet $LIST; fi"
+    "lint:changed": "LIST=`git diff-index --name-only HEAD | grep \"\\.js[x]\\{0,1\\}$\"`; if [ \"$LIST\" ]; then eslint --fix --quiet $LIST; fi",
+    "prepare": "husky install"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "yarn lint:fix && yarn build"
+    }
+  },
```
Comment (theodur) [inline package.json:23]: Do we need this since we already have the `pre-commit` script in the husky directory?

File: package.json [hunk 2]
```diff
@@ -51,5 +62,6 @@
     "**/diff": "^3.5.0",
     "**/growl": "1.10.2",
     "**/minimist": "^1.2.3"
-  }
+  },
+  "dependencies": {}
```
Comment (theodur) [inline package.json:66]: Was this added for a specific reason?

File: package.json (hunk 1 above)
Comment (LinnJS) [reply package.json:23]: You are right, this is from the old version of Husky

File: package.json (hunk 2 above)
Comment (LinnJS) [reply package.json:66]: No idea how this got there will revert, good catches 🙂

Comment (LinnJS) [reply package.json:66]: Oh looks like this got there from my accidentally adding husky as a `dependency` and not a `devDependency`

**Comment by LinnJS** [comment]
I fixed all ESLint errors other than the ones below since they directly effected the schema `dist` and property names. I feel that these should be fixed by those application owners or each should be done in separate PRs with property references checked on FE and BE on every PR.

image

**Review by craigwheeler — APPROVED** [review]

**Review by theodur — APPROVED** [review]
This looks good to me! Thanks for making these changes 🚀

**Comment by LinnJS** [comment]
@theodur no probelm just want to make all our lives easier, thanks for being open to the idea 🥇

**Pull request merged**